packages/edit-navigation/src/components/menu-editor/use-navigation-blocks.js

```diff
@@ -14,6 +14,7 @@ function createBlockFromMenuItem( menuItem ) {
 	return createBlock( 'core/navigation-link', {
 		label: menuItem.title.raw,
 		url: menuItem.url,
+		id: menuItem.id,
```

Comment:
I think this is a mistake, because I am also working on the dynamic links PR which introduces id in the block. Not sure that is the reason but good catch!